### PR TITLE
Fix host and port for proxy connections

### PR DESCRIPTION
**Pull request by samuelfekete**
This fixes errors when using a proxy, as described in this issue:
https://github.com/kevin1024/vcrpy/issues/214

### CODE REVIEW

**Comment by kevin1024** [comment]
Thanks.  Please provide a test that shows the bug this is fixing.

**Comment by graingert** [comment]
I'd also consider patching urllib3 rather than making our socket mock falsy

**Comment by samuelfekete** [comment]
@graingert, it appears that urllib3 does that for a reason, so I wouldn't change that.

Instead, I changed VCRConnection to only have a `sock` once a connection has been made. urllib3 is expecting this behaviour because that is what the httplib HTTPConnection does: https://github.com/python/cpython/blob/2.7/Lib/httplib.py#L820

File: vcr/stubs/__init__.py [hunk 1]
```diff
@@ -136,6 +136,9 @@ def _port_postfix(self):
 
     def _uri(self, url):
         """Returns request absolute URI"""
+        if url and not url.startswith('/'):
+            # Then this must be a proxy request.
+            return url
```
Comment (hartwork) [inline vcr/stubs/__init__.py]: 👍 That fixes handling with `http_proxy` in the environment, as well.

**Comment by Smosker** [comment]
How things are going with this PR? i got same issue.

**Comment by samuelfekete** [comment]
@Smosker, this fixes the issue, but tests are failing because they are failing on master (https://github.com/kevin1024/vcrpy/issues/338).

I am currently not working on this, so you are welcome to have a go at getting the tests to pass.

**Comment by hartwork** [comment]
@samuelfekete I'm optomistic that with the latest merge to `master` [the test suite](https://travis-ci.org/kevin1024/vcrpy/builds/329568738) will return to zero failures. Please rebase this pull request against latest `master`. Thank you!

**Comment by lamenezes** [comment]
hello @samuelfekete, thanks for your PR. can you update your branch? tests have been fixed and are working normally again :soccer: :top:

**Comment by samuelfekete** [comment]
@hartwork, I have rebased and it's green! Thanks @lamenezes for the reminder.

**Comment by lamenezes** [comment]
@samuelfekete thanks for updating this PR. I will review this as soon as I can :)

**Comment by Iftahh** [comment]
@lamenezes any update on this?
I'm waiting for the proxy patch to be merged so I can use the wonderful vcrpy at work.

**Pull request merged**